src/analyserexternalvariable.cpp

```diff
@@ -156,21 +152,15 @@ VariablePtr AnalyserExternalVariable::dependency(const ModelPtr &model,
     auto result = mPimpl->findDependency(model, componentName, variableName);
 
     if (result != mPimpl->mDependencies.end()) {
-        return result->lock();
+        return *result;
     }
 
     return nullptr;
```

Comment:
Is `return (result != mPimpl->mDependencies.end()) ? *result : nullptr;` cleaner?

Reply:
1def632e